python/ray/tests/test_runtime_env_conda_and_pip_4.py

```diff
@@ -112,5 +112,23 @@ def test_run_in_virtualenv(cloned_virtualenv):
     cloned_virtualenv.run(f"IN_VIRTUALENV=1 python -m pytest {__file__}", capture=True)
 
 
+@pytest.mark.skipif(
+    "IN_VIRTUALENV" in os.environ or sys.platform != "linux",
+    reason="Requires PR wheels built in CI, so only run on linux CI machines.",
```

Comment:
Can we run all these new tests on Mac as well, and update the `reason` field?  Actually, only `conda` tests need wheels to be built from the PR, because for `conda` we install an isolated environment and we need to put the Ray wheel in there.  The `pip` tests should be fine.

In the future we should also enable these on Windows, but we will need to wait until https://github.com/ray-project/ray/pull/22449 is merged first.